### PR TITLE
restructure dataflow component structure

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/338)

### CODE REVIEW

File: components/dataflow/predict/build_image.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ fi
 
 # build base image
 pushd ../base
-./build.sh
+build_image.sh
```
Comment (Ark-kun) [inline components/dataflow/predict/build_image.sh]:
```suggestion
./build_image.sh
```

File: components/dataflow/tfdv/build_image.sh [hunk 2]
```diff
@@ -44,7 +44,7 @@ fi
 
 # build base image
 pushd ../base
-./build.sh
+build_image.sh
```
Comment (Ark-kun) [inline components/dataflow/tfdv/build_image.sh]:
```suggestion
./build_image.sh
```

File: components/dataflow/tfma/build_image.sh [hunk 3]
```diff
@@ -44,7 +44,7 @@ fi
 
 # build base image
 pushd ../base
-./build.sh
+build_image.sh
```
Comment (Ark-kun) [inline components/dataflow/tfma/build_image.sh]:
```suggestion
./build_image.sh
```

File: components/dataflow/tft/build_image.sh [hunk 4]
```diff
@@ -44,7 +44,7 @@ fi
 
 # build base image
 pushd ../base
-./build.sh
+build_image.sh
```
Comment (Ark-kun) [inline components/dataflow/tft/build_image.sh]:
```suggestion
./build_image.sh
```

File: components/dataflow/base/build_image.sh [hunk 5]
```diff
@@ -15,12 +15,12 @@
 
 
 mkdir -p ./build
-rsync -arvp "../../tfdv"/ ./build/
-rsync -arvp "../../tft"/ ./build/
-rsync -arvp "../../tfma"/ ./build/
-rsync -arvp "../../predict"/ ./build/
-cp ../../../license.sh ./build
-cp ../../../third_party_licenses.csv ./build
+rsync -arvp "../tfdv/src"/ ./build/
```
Comment (Ark-kun) [inline components/dataflow/base/build_image.sh]: In future we should probably try to separate the component images. For now it's fine.

Comment (gaoning777) [reply components/dataflow/base/build_image.sh]: are you talking about separating the component directory from the pipeline repo?

Comment (Ark-kun) [reply components/dataflow/base/build_image.sh]: No. I'm talking about having one image per dataflow component. Right now they're using a single image. But this is low-priority.

**Comment by Ark-kun** [comment]
Are any sample test using these components?

**Comment by gaoning777** [comment]
Yes, all these components are tested by the sample tests.

**Comment by gaoning777** [comment]
Tested on local cloudbuild; it works fine.

File: components/dataflow/base/build_image.sh (hunk 5 above)
Comment (qimingj) [reply components/dataflow/base/build_image.sh]: We do have one image per dataflow component. This is the base image that captures the dependencies. The source code is built into leaf images.

**Pull request merged**